miio/cooker_multi.py

```diff
@@ -0,0 +1,419 @@
+        self.profile_bytes[15] = (duration % 60 | self.profile_bytes[15] & 0x80) & 0xFF
+
+        self.update_checksum()
+
+    def is_akw_enabled(self):
+        return (self.profile_bytes[15] & 0x80) == 0x80
+
+    def set_akw_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[15] |= 0x80
+        else:
+            self.profile_bytes[15] &= 0x7F
+
+        self.update_checksum()
+
+    def calc_checksum(self):
+        crc = crcmod.mkCrcFun(0x11021, rev=False, initCrc=0x0, xorOut=0x0)(
+            self.profile_bytes
+        )
+        checksum = bytearray(2)
+        checksum[0] = (crc >> 8) & 0xFF
+        checksum[1] = crc & 0xFF
+        return checksum
+
+    def update_checksum(self):
+        self.checksum = self.calc_checksum()
+
+    def is_valid(self):
+        return len(self.profile_bytes) == 174 and self.checksum == self.calc_checksum()
+
+    def get_profile_hex(self):
+        return (self.profile_bytes + self.checksum).hex()
+
+
+class CookerStatus(DeviceStatus):
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["status"])
+
+    @property
+    def menu(self) -> str:
+        """Selected menu id."""
+        try:
+            return COOKING_MENUS[self.data["menu"]]
+        except KeyError:
+            return "Unknown menu"
+
+    @property
+    def stage(self) -> str:
+        """Current stage if cooking."""
+        try:
+            return COOKING_STAGES[self.data["phase"]]["name"]
+        except KeyError:
+            return "Unknown stage"
+
+    @property
+    def temperature(self) -> Optional[int]:
+        """Current temperature, if idle.
+
+        Example values: 29
+        """
+        return self.data["temp"]
+
+    @property
+    def start_time(self) -> int:
+        """Start time of cooking?"""
+        return int(self.data["t_start"])
+
+    @property
+    def remaining(self) -> int:
+        """Remaining minutes of the cooking process."""
+        return int(int(self.data["t_left"]) / 60)
+
+    @property
+    def cooking_delayed(self) -> Optional[int]:
+        """Wait n minutes before cooking / scheduled cooking."""
+        delay = int(self.data["t_pre"])
+
+        if delay >= 0:
+            return delay
+
+        return None
+
+    @property
+    def duration(self) -> int:
+        """Duration of the cooking process."""
+        return int(self.data["t_cook"])
+
+    @property
+    def keep_warm(self) -> bool:
+        """Keep warm after cooking?"""
+        return self.data["akw"] == 1
+
+    @property
+    def settings(self) -> None:
+        """Settings of the cooker."""
+        return None
+
+    @property
+    def hardware_version(self) -> None:
+        """Hardware version."""
+        return None
+
+    @property
+    def firmware_version(self) -> None:
+        """Firmware version."""
+        return None
```

Comment:
I'd remove these if you are not planning to implement them.

Reply:
done